### PR TITLE
Update distribution.yaml: add rosx_introspection

### PR DESCRIPTION
**Pull request by facontidavide**
Please add the following dependency to the rosdep database.

## Package name:

rosx_introspection

## Package Upstream Source:

https://github.com/facontidavide/rosx_introspection

## Purpose of using this:

Needed by **foxglove_bridge**

## Links to Distribution Packages

Not yet

# Please Add This Package to be indexed in the rosdistro.

rosx_introspection

# The source is here:

https://github.com/facontidavide/rosx_introspection.git

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: humble/distribution.yaml [hunk 1]
```diff
@@ -7985,6 +7985,16 @@ repositories:
       url: https://github.com/DFKI-NI/rospy_message_converter.git
       version: humble
     status: maintained
+  rosx_introspection:
+    doc:
+      type: git
+      url: https://github.com/facontidavide/rosx_introspection.git
+      version: main
```
Comment (audrow) [inline humble/distribution.yaml:7992]: It seems you have a `master` branch and not `main` so CI is failing.

Comment (facontidavide) [inline humble/distribution.yaml:7992]:
```suggestion
      version: master
```

**Comment by facontidavide** [comment]
fixed. Thanks

File: humble/distribution.yaml [hunk 2]
```diff
@@ -7985,6 +7985,16 @@ repositories:
...
     status: maintained
+  rosx_introspection:
+    doc:
+      type: git
+      url: https://github.com/facontidavide/rosx_introspection.git
+      version: master
+    source:
+      type: git
+      url: https://github.com/facontidavide/rosx_introspection.git
+      version: main
```
Comment (facontidavide) [inline humble/distribution.yaml:7996]:
```suggestion
      version: master
```

**Comment by marcoag** [comment]
[Holding for Humble sync](https://discourse.ros.org/t/preparing-for-humble-sync-2024-06-19/38236)

**Review by audrow — APPROVED** [review]

**Pull request merged**